File: sol/2410/2410.cpp
struct Solution {
    int matchPlayersAndTrainers(vector<int>& players, vector<int>& trainers) {
        ranges::sort(players);
        ranges::sort(trainers);
        int n = trainers.size(), ptr = 0, res = 0;
        for (int p : players){
            while (ptr != n && trainers[ptr] < p){
                ++ptr;
            }
            if (ptr == n){
                break;
            }
            ++res;
            ++ptr;
        }
        return res;
    }
};

File: sol/2410/2410.py
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        players.sort()
        trainers.sort()
        n = len(trainers)
        ptr = 0
        res = 0
        for p in players:
            while ptr != n and trainers[ptr] < p:
                ptr += 1
            if ptr == n:
                break
            res += 1
            ptr += 1
        return res
